### ISSUE TITLE
Python data values, part 3 continued: invert code in order to pass values back from Python to a module--SparseRowMatrix

### ISSUE THREAD

**Issue by dcwhite** [SCIInstitute/SCIRun#1299]

**Comment by dcwhite** [comment]
@jessdtate Could you make an example for this one?

**Comment by jessdtate** [comment]
Passing sparse back from IWP seems to be working on a pass through.  Is there an API to send python data to a scirun module port?

**Comment by dcwhite** [comment]
Not really. How do you want that to work? The layer that translates between data would need to operate somehow "above" the network.

**Comment by jessdtate** [comment]
I just keep getting confused which format the data is in, and I've never been able to sucessfully compile a sparse matrix, either from scratch or an existing dense matrix (from scipy or another package.   What would be helpful from a user perspective is a description of the format that we use (from eigen, but perhaps with some clearer examples) and maybe a converter from scipy sparse to scirun sparse, or an example.

**Comment by dcwhite** [comment]
https://eigen.tuxfamily.org/dox/group__TutorialSparse.html

https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csc_matrix.html#scipy.sparse.csc_matrix

These two formats might be about the same. Good intern task to figure out converting them, and writing up some helpful documentation...

**Comment by dcwhite** [comment]
**Triage 2026-06:** Python data values part 3 — `SparseRowMatrix` Python↔SCIRun conversion (invert direction: Python values back to module). The dense matrix and column matrix conversions exist. `SparseRowMatrix` requires converting from scipy.sparse format. Medium effort; the conversion code is in `src/Modules/Python/`.
